components/v1/Buttons/ConnectWalletDesktop.tsx

```diff
@@ -82,6 +82,19 @@ const ButtonConnectWalletDesktop: FunctionComponent<ButtonConnectWalletDesktopPr
         setConnectorName(c.name);
 
         const result = await connect(c);
+
+        // Prevent connecting with WalletConnect if network is not right
+        if(c instanceof WalletConnectConnector){
+            if(result?.data?.chain?.unsupported || result?.data?.chain?.id !== chain.id){
```

Comment:
@math-marcellino btw If we use this `result?.data?.chain?.id`, we can just use one `WCConnector` like before tho

Reply:
oh okay sir, I think I forgot that the one who automatically disconnect when the network is not supported & chainID not defined in WCConnector is useDApp not Wagmi xD, let me fix this again, Thank you!